sched/init/nx_start.c

```diff
@@ -762,12 +762,6 @@ void nx_start(void)
 
   DEBUGASSERT(this_cpu() == 0 && CONFIG_MAX_TASKS > CONFIG_SMP_NCPUS);
 
-  /* Take the memory manager semaphore on this CPU so that it will not be
-   * available on the other CPUs until we have finished initialization.
-   */
-
-  DEBUGVERIFY(kmm_trysemaphore());
```

Comment:
Taking this semaphore is not necessary?

Reply:
Yes, because the heap manager is fully workable in this stage.